app/controllers/hyrax/file_sets_controller.rb

```diff
@@ -147,11 +153,24 @@ def search_builder_class
 
     def initialize_edit_form
       @parent = @file_set.in_objects.first
+      workflow_check(parent_id: @parent.id)
       original = @file_set.original_file
       @version_list = Hyrax::VersionListPresenter.new(original ? original.versions.all : [])
       @groups = current_user.groups
     end
 
+    def workflow_check(parent_id: nil)
```

Comment:
I'm contemplating the value of extracting this as a Service class.

Reply:
i feel like it would really help to extract this to a service. not least: it would make the exact behavior of this check easier to test robustly and and much easier to change later if we didn't get it perfect this time.